conda/base/context.py

```diff
@@ -822,6 +822,15 @@ def user_agent(self):
         builder.append("%s/%s" % self.os_distribution_name_version)
         if self.libc_family_version[0]:
             builder.append("%s/%s" % self.libc_family_version)
+        if self.experimental_solver.value != "classic":
+            from ..core.solve import _get_solver_class
+
+            user_agent_str = "solver/%s" % self.experimental_solver.value
+            Solver = _get_solver_class()
+            if hasattr(Solver, "user_agent") and callable(Solver.user_agent):
+                # This has to be a static or class method
+                user_agent_str += f" {Solver.user_agent()}"
```

Comment:
```suggestion
            try:
                # Solver.user_agent has to be a static or class method
                user_agent_str += f" {Solver.user_agent()}"
            except (AttributeError, TypeError):
                pass
```

Reply:
The solver class might need to compute things on the fly (like the version of a library it depends on), hence the callable approach. I'll go EAFP for this!